sdcm/utils/k8s/chaos_mesh.py

```diff
@@ -44,3 +67,106 @@ def initialize(self) -> None:
             timeout="30m"
         )
         LOGGER.info("chaos-mesh installed successfully on %s k8s cluster.", self._k8s_cluster.k8s_scylla_cluster_name)
+
+
+class ExperimentStatus(Enum):
+    STARTING = 0
+    RUNNING = 1
+    PAUSED = 2
+    FINISHED = 3
+    ERROR = 4
+    UNKNOWN = 5
+
+
+class PodChaosExperiment:
+    """Base class for all PodChaos experiments."""
+    API_VERSION = "chaos-mesh.org/v1alpha1"
+
+    def __init__(self, pod: "BasePodContainer", name: str):
+        self._k8s_cluster = pod.parent_cluster.k8s_cluster
+        self._name = name
+        self._namespace = pod.parent_cluster.namespace
+        self._experiment = {
+            "apiVersion": self.API_VERSION,
+            "metadata": {
+                "name": self._name,
+                "namespace": self._namespace
+            }
+        }
+
+    def start(self):
+        """Starts experiment. Does not wait for finish."""
+        LOGGER.debug("Starting a pod-failure experiment %s", self._name)
+        assert self._k8s_cluster, "K8s cluster hasn't been configured for this experiment."
+        with NamedTemporaryFile(suffix=".yaml", mode="w") as experiment_config_file:
+            yaml.dump(self._experiment, experiment_config_file)
+            experiment_config_file.flush()
+            self._k8s_cluster.apply_file(experiment_config_file.name)
+        LOGGER.info("pod-failure experiment '%s' has started", self._name)
+
+    def get_status(self) -> ExperimentStatus:
+        """Gets status of podchaos experiment."""
+        result = self._k8s_cluster.kubectl(
+            f"get podchaos {self._name}  -n {self._namespace} -o jsonpath='{{.status.conditions}}'", verbose=False)
+        condition = {cond["type"]: ast.literal_eval(cond["status"]) for cond in json.loads(result.stdout)}
+        if not condition["Selected"] and condition["Paused"]:
+            return ExperimentStatus.ERROR
+        if not condition["Selected"] and not condition["Paused"] and not condition["AllRecovered"] and not condition["AllInjected"]:
```

Comment:
It may be shorter:
```
if not (condition["Selected"] or condition["Paused"] or condition["AllRecovered"] or condition["AllInjected"]):
```

The same is valid also in other conditions.

Reply:
I rather had all conditionals to be in the same form, where for each condition we use `and` or not use condition at all when has no impact on given status. For me it is more readable. I don't care about extra few characters.

Reply:
The proposed variant is also simple.

Reply:
but is different from the others - so will require brain creating new pattern to compile and use taking cycles, memory and context switch.

And finally I don't see it as important - this is just implementation detail.

Reply:
not critical change from my POV